### ISSUE TITLE
[Modulo 3] - [Aula 8] - Exercício 3

### ISSUE THREAD

**Issue by bia-diniz** [cwi-reset/edicao-04-level-1#325]
Olá, pessoal.
Fazendo os exercícios me deparei com um possível erro na resposta  da exercício 3 da aula 8.
Após rodar o código só imprimi na tela a mensagem "é nome comum", mesmo não sendo um nome presente no array.
E meu código esta exatamente igual a resposta proposta pela CWI.
Alguém consegue me dar uma luz? haha..

**Comment by Askalor** [comment]
> Olá, pessoal. Fazendo os exercícios me deparei com um possível erro na resposta da exercício 3 da aula 8. Após rodar o código só imprimi na tela a mensagem "é nome comum", mesmo não sendo um nome presente no array. E meu código esta exatamente igual a resposta proposta pela CWI. Alguém consegue me dar uma luz? haha..

Se você rodar o código da CWI, ele dá certo? Você pode tentar limpar o cache do navegador e rodar os códigos novamente. Se ainda sim não funcionar, poste o código aqui.

**Comment by bia-diniz** [comment]
Eu não consegui. Estou usando o VSCode. 
Segue meu código.

nomesComuns = ["Miguel", "Laura", "Lucas", "Beatriz", "Guilherme", "Maria", "Gabriel", "Ana", "Arthur", "Júlia", 
"Enzo", "Alice", "Rafael", "Mariana", "João", "Larissa", "Gustavo", "Maria Eduarda", "Pedro", "Sofia", 
"Bernardo", "Isabela", "Matheus", "Helena", "Davi", "Camila", "Heitor", "Lara", "Henrique", "Valentina", 
"Bruno", "Letícia", "Samuel", "Luana", "Felipe", "Amanda", "Lorenzo", "Yasmin", "Benjamin", "Sophia", 
"Vinícius", "Rebeca", "Rodrigo", "Juliana", "Eduardo", "Bruna", "Diego", "Cecília", "Antônio", "Fernanda", 
"Leonardo", "Isadora", "Noah", "Lorena", "Nícolas", "Lívia", "Daniel", "Manuela", "Thiago", "Vitória"]

meuNome = "Bia"

index = 0
for(index = 0 ; index < nomesComuns.length ; index++){
    if (meuNome == nomesComuns[index]){
        nomesComuns = true
        break
    }
} 

if (nomesComuns){
    console.log ("É, nome comum :P")
} else {
    console.log ("Diferentão, hein? XD")

**Comment by Askalor** [comment]
> Eu não consegui. Estou usando o VSCode. Segue meu código.
> 
> nomesComuns = ["Miguel", "Laura", "Lucas", "Beatriz", "Guilherme", "Maria", "Gabriel", "Ana", "Arthur", "Júlia", "Enzo", "Alice", "Rafael", "Mariana", "João", "Larissa", "Gustavo", "Maria Eduarda", "Pedro", "Sofia", "Bernardo", "Isabela", "Matheus", "Helena", "Davi", "Camila", "Heitor", "Lara", "Henrique", "Valentina", "Bruno", "Letícia", "Samuel", "Luana", "Felipe", "Amanda", "Lorenzo", "Yasmin", "Benjamin", "Sophia", "Vinícius", "Rebeca", "Rodrigo", "Juliana", "Eduardo", "Bruna", "Diego", "Cecília", "Antônio", "Fernanda", "Leonardo", "Isadora", "Noah", "Lorena", "Nícolas", "Lívia", "Daniel", "Manuela", "Thiago", "Vitória"]
> 
> meuNome = "Bia"
> 
> index = 0 for(index = 0 ; index < nomesComuns.length ; index++){ if (meuNome == nomesComuns[index]){ nomesComuns = true break } }
> 
> if (nomesComuns){ console.log ("É, nome comum :P") } else { console.log ("Diferentão, hein? XD")

Então, o erro está no seu loop. Você está atribuindo ao Array `nomesComum`  a condição `true`, e então comparando no if. Você precisa de uma outra variável para fazer isso.

```
for(index = 0 ; index < nomesComuns.length ; index++){
if (meuNome == nomesComuns[index]){
nomesComuns = true // Essa variável é seu array, troque por outra e adicione-a no if
break
}
}
```

**Comment by BWartchow** [comment]
sim, além disso, acho que precisaria antes do for declarar a variável como false, né? Pq ela só vira true se encontrar um nome igual

**Comment by bia-diniz** [comment]
Obrigada, pessoal! 
Realmente só faltou declarar o false fora do for.
Agora deu certinho aqui!

Mas isso realmente não aparece na resposta da CWI.

**Comment by Askalor** [comment]
> Obrigada, pessoal! Realmente só faltou declarar o false fora do for. Agora deu certinho aqui!
> 
> Mas isso realmente não aparece na resposta da CWI.

De nada

Na resposta deles, eles usaram variáveis com nomes semelhantes. `nomesComuns` para o array e `nomeComum` para a booleana. E ela foi declarada como falsa logo na segunda linha do código.

**Comment by bia-diniz** [comment]
Caramba! Eu juro que não estou vendo isso. haha
Fiquei quebrando a cabeça aqui. 
Mas estou feliz por ter conseguido agora!

**Comment by Askalor** [comment]
Bastante estranho mesmo. Aqui está assim, logo na segunda linha, antes dos nomes: 
![image](https://user-images.githubusercontent.com/112364290/190938662-43f10e3f-ce64-4ff1-ad6f-ae2b6ccf8f8a.png)

Mas boa sorte com os próximos exercícios e bons estudos!

**Comment by bia-diniz** [comment]
Poxa, pessoal! O problema era eu mesmo.
Acho q o cansaço tinha batido ontem. Hahaha..
Obrigada pela ajuda e desculpa a comida de bola. 😅
